### PR TITLE
Add Experience Builder

### PR DESCRIPTION
**Pull request by phuongnq**
https://github.com/craftercms/craftercms/issues/2518

### CODE REVIEW

File: sources/packages/app/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+              >
+                Learn React
+              </a>
+              <div>
+                <div className="App-note">
+                <RenderField
+                    model={model}
+                    fieldId="body_html"
+                    render={(value, fieldId) => (<div dangerouslySetInnerHTML={{ __html: value }} />) }
+                  />
```
Comment (rart) [inline sources/packages/app/src/App.tsx]:
```suggestion
                <RenderField
                    model={model}
                    fieldId="body_html"
                    renderTarget="dangerouslySetInnerHTML"
                    render={(value) => ({ __html: value }) }
                  />
```

File: sources/packages/app/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+              <RenderField
+                model={model}
+                fieldId="headText_s"
+                component="p"
+              />
+              <RenderField
+                model={model}
+                fieldId="logo_s"
+                render={(logo, fieldId) => (<img src={`${BASE_URL}${logo}`} className="App-logo" alt="logo" />)}
+              />
```
Comment (rart) [inline sources/packages/app/src/App.tsx]:
```suggestion
              <RenderField
                model={model}
                fieldId="logo_s"
                renderTarget="src"
                component="img"
                className="App-logo"
                alt="logo"
                render={(logo) => `${BASE_URL}${logo}`}
              />
```

File: sources/packages/app/src/index.tsx [hunk 3]
```diff
@@ -1,8 +1,24 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
+import { crafterConf } from '@craftercms/classes';
+// @ts-ignore
+import cookies from 'js-cookie';
```
Comment (rart) [inline sources/packages/app/src/index.tsx]: Isn't there a `yarn add @types/js-cookie -D` that fixes this?

File: templates/web/myapp.ftl
Comment (rart) [inline templates/web/myapp.ftl:1]: Rename ftl to `app.ftl`

File: config/studio/content-types/page/myapp/form-definition.xml
Comment (rart) [inline config/studio/content-types/page/myapp/form-definition.xml:1]: Can we rename this content type too to something other than `myapp`? Even if it is _Example_.

**Review by rart — CHANGES_REQUESTED** [review]
- Migrate to Vite.
- Might have been GitHub but format looked off while checking. Should run prettier for all to make sure.

File: sources/packages/app/src/App.tsx (hunk 1 above)
Comment (phuongnq) [reply sources/packages/app/src/App.tsx]: Updated as suggested

File: sources/packages/app/src/index.tsx (hunk 3 above)
Comment (phuongnq) [reply sources/packages/app/src/index.tsx]: Added `@types/js-cookie`

File: templates/web/myapp.ftl
Comment (phuongnq) [reply templates/web/myapp.ftl:1]: Updated

File: config/studio/content-types/page/myapp/form-definition.xml
Comment (phuongnq) [reply config/studio/content-types/page/myapp/form-definition.xml:1]: I updated to `/page/app`

File: sources/packages/app/src/App.tsx (hunk 2 above)
Comment (phuongnq) [reply sources/packages/app/src/App.tsx]: @rart I have tried the following, but the XB is not enabled while I hover the cursor (other components are working fine, the current code is also working fine). Let me know if there is something wrong with the code.
```
<RenderField
                model={model}
                fieldId="logo_s"
                renderTarget="src"
                component="img"
                componentProps={{
                  className: "App-logo",
                  alt: "logo"
                }}
                render={(logo) => `${BASE_URL}${logo}`}
              />
```

Comment (rart) [reply sources/packages/app/src/App.tsx]: That's strange sounds like a bug. Will have to figure out why the img isn't working as suggested (@jvega190 please investigate this issue)

Does the drag & drop of image work well — both from desktop and assets panel — rendering in this way?

FYI: There's a typings issue where Typescript won't accept the extra props at the root (e.g. className, alt) but they are actually forwarded and do work even out of the `componentProps`.

**Review by rart — APPROVED** [review]

File: sources/packages/app/src/App.tsx (hunk 2 above)
Comment (jvega190) [reply sources/packages/app/src/App.tsx]: Hi @phuongnq , the issue was caused by a style applied to the image (with class `App-logo`). There was a `pointer-events: none` style that caused the hover and click (and others) not to work properly. You can take a look at https://github.com/craftercms/react-blueprint/pull/2/files#diff-7f8d10e8f03b240d4556ab4e56fcdf069d1be4b7f0d13ae7c5d6326d54d4bf1cL7

**Comment by jvega190** [comment]
@phuongnq I made some updates to the bp, you can see them in [here](https://github.com/craftercms/react-blueprint/pull/2#). 
- I updated the image-picker and rte controls for them to have the image datasources linked 
- I also added some logic for the image not to spin while editMode is on

**Pull request merged**